client/organization_factory.go

```diff
@@ -192,27 +217,61 @@ func (f *OrganizationFactory) getCredentialSecret(ctx context.Context, objectMet
 		return nil, microerror.Mask(tooManyCredentialsError)
 	}
 
+	if len(secretList.Items) < 1 {
+		return nil, microerror.Mask(credentialsNotFoundError)
+	}
+
 	// If one credential secret is found, we use that.
-	if len(secretList.Items) == 1 {
-		secret := secretList.Items[0]
+	secret := secretList.Items[0]
 
-		credentialSecret := &v1alpha1.CredentialSecret{
-			Namespace: secret.Namespace,
-			Name:      secret.Name,
+	credentialSecret := &v1alpha1.CredentialSecret{
+		Namespace: secret.Namespace,
+		Name:      secret.Name,
+	}
+
+	f.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found credential secret %s/%s", credentialSecret.Namespace, credentialSecret.Name))
+
+	return credentialSecret, nil
+}
+
+// getLegacyCredentialSecret tries to find a Secret in the default credentials namespace but labeled with the organization name.
+// This is needed while we migrate everything to the org namespace and org credentials are created in the org namespace instead of the default namespace.
+func (f *OrganizationFactory) getLegacyCredentialSecret(ctx context.Context, objectMeta v1.ObjectMeta) (*v1alpha1.CredentialSecret, error) {
```

Comment:
`getOrganizationCredentialSecret` and `getLegacyCredentialSecret` look the same, maybe use one function with namespace as a parameter. Not a blocker.